Source/WebKit/UIProcess/API/C/WKAXTypes.h

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (C) 2023 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef WKAXTypes_h
+#define WKAXTypes_h
+
+#include <WebKit/WKBase.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum AXNotification {
+    // must be same as WebCore::AXObjectCache::AXNotification
```

Comment:
Basically this file is my main point of concern wrt review feedback.

The most useful aspect of us upstreaming this code is to ensure that we don't keep getting broken every time AX enums change, so the solution here is to introduce an "everybody file" such that no platform is able to update the WebCore enums without keeping WKAPICast in sync.

But this kinda sucks if no platform other than PS is likely to actually _use_ the casts. What I really want to do is do is JSC would do: have WebCore define these enums by means of an X macro pattern. So in AXObjectCache.h we'd have...
```cpp
#define WEBCORE_AXNOTIFICATION_KEYS(macro)  \
    macro(AccessKeyChanged)
    ...
```
...which would be used to define `WebCore::AXNotification` in the same way as [`JSC::TemporalUnit`](https://github.com/WebKit/WebKit/blob/main/Source/JavaScriptCore/runtime/TemporalObject.h#L45-L57), and which could then be used in WebKit by any specific platforms that wish to generate API casts for the same enum values.

If we were okay with that idea, I'd be willing to do so as a separate patch prior to landing this one.